.github/workflows/android_deploy_to_production.yml

```diff
@@ -0,0 +1,68 @@
+name: android_deploy_to_production
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+  # push:
+  #   branches:
+  #     - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      image_ref: ${{ vars.DOCKER_REGISTRY }}/${{ vars.DOCKER_USERNAME }}/boilerplate-react-native@${{ steps.build.outputs.image_digest }}
+    steps:
+      - name: Checkout (app)
+        uses: actions/checkout@v3
+        # make sure to keep checked in code at different directory to avoid colliding with docker caching
+        with:
+          path: app
+
+      - name: Extract branch name
+        id: extract_branch
+        shell: bash
+        run: |
+          BRANCH_NAME=$(echo ${{ github.event.pull_request.head.ref }} | sed -e 's/^refs\/heads\///g')
+          BRANCH_HASH=$(sha1sum < <(printf '%s' $BRANCH_NAME) | cut -c -15)
+
+          echo "##[set-output name=branch_hash;]$(echo $BRANCH_HASH)"
+
+      - name: Build application
+        id: build
+        uses: ./app/.github/actions/build
+        with:
+          tags: ${{ vars.DOCKER_REGISTRY }}/${{ vars.DOCKER_USERNAME }}/boilerplate-react-native:${{ steps.extract_branch.outputs.branch_hash }}
+          build_args: NODE_CONFIG_ENV=production
+          build_secrets: ''
+          context: app/
+          docker_registry: ${{ vars.DOCKER_REGISTRY }}
+          docker_username: ${{ vars.DOCKER_USERNAME }}
+          docker_password: ${{ secrets.DOCKER_PASSWORD }}
+  
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v2 
+      - name: Decrypt Android keys 
+        run: sh ./.github/scripts/decrypt_android_keys.sh 
+        env: 
+          RELEASE_KEYSTORE: ${{ secrets.RELEASE_KEYSTORE }}
+          RELEASE_KEYSTORE_PASSPHRASE: ${{ secrets.RELEASE_KEYSTORE_PASSPHRASE }}
+          SERVICE_ACCOUNT: ${{ secrets.SERVICE_ACCOUNT }}
+          SERVICE_ACCOUNT_PASSPHRASE: ${{ secrets.SERVICE_ACCOUNT_PASSPHRASE }}
+   
+      - uses: actions/setup-ruby@v1 
+        with: 
+         ruby-version: '3.x' 
+   
+      - name: Install Fastlane
+        run: gem install fastlane
+
+      - name: Upload to PlayStore 
+        run: | 
+            cd android 
+            fastlane android release_production RELEASE_KEYSTORE_PASSWORD:${{secrets.RELEASE_KEYSTORE_PASSWORD}} RELEASE_KEYSTORE_KEY_PASSWORD:${{secrets.RELEASE_KEYSTORE_KEY_PASSWORD}} RELEASE_KEYSTORE_ALIAS:${{secrets.RELEASE_KEYSTORE_ALIAS}}
```

Comment:
Following comments:
- Can ruby and faslane be part of the docker image itself? Ideally, our entire env should be within docker container.
- Ideally you get away with **not using** `actions/checkout@v2` as you have everything required within your docker image.
- Then you actually need to either:
  - Pull down build image and run the fastlane command against it. The entire ecosystem is within the image and you'd be able to run the command if ruby or fastlane is properly setup.
  - Github also supports running the entire job within a container which can be built using image of your choice. You can supply the path to the image we have built, Github should be able to pull down the image, run a container. You then, theoretically, don't need to run commands against the docker image. You should be simply able to do `fastlane android ....` as you are within the container.